### ISSUE TITLE
[Question] Replace Fn and Ctrl key

### ISSUE THREAD

**Issue by bt** [daliansky/XiaoMi-Pro-Hackintosh#250]
Is there a way I can replace the functionality of the Fn and Ctrl key on the left of the keyboard? Similar to the BIOS setting on Lenovo laptops. 

This is to make the Mi Notebook more aligned with the keyboard style of the MacBook, where Fn is on the bottom left corner of the keyboard compared to Ctrl being on the bottom left.

**Comment by stevezhengshiqi** [comment]
You can remap ADB code of Ctrl key to let it become a Fn.
If you can find the PS2 code of Fn key, then you can remap Ctrl function to the Fn key.
Ref: https://github.com/daliansky/XiaoMi-Pro-Hackintosh/issues/181

**Comment by bt** [comment]
Thanks for the reply @stevezhengshiqi !

I have looked at your reference issue, but seems like you were unable to find the PS2 code for Fn key? I think this is common on most laptops containing an Fn key unfortunately; where Fn+Fx is required to issue a code :(

**Comment by stevezhengshiqi** [comment]
Yeah, so that issue is help wanted. I can’t find PS2 code of some keys. They may controlled by GPE queries. It would be awesome of you can find it.

**Comment by Menchen** [comment]
I think fn and toggle touchpad are handled directly by EC, that explains why they sork without driver or mapping in linux, and fn lock persists across multiple os.

**Comment by bt** [comment]
@Menchen yeah, I think so. I'm unsure if we can patch the ACPI to swap the keys around, might be a bit of work there.

@stevezhengshiqi I tried already using Karabiner, but it doesn't detect the press of Fn key, until you combo it with another key.

**Comment by stevezhengshiqi** [comment]
Actually, Xiaomi-Pro doesn't have a EC device in ACPI or EmbeddedControl field in ACPI. The Fn key and so on are handled by GPE query.
